arci/src/clients/partial_joint_trajectory_client.rs

```diff
@@ -30,13 +30,59 @@ impl<C> PartialJointTrajectoryClient<C>
 where
     C: JointTrajectoryClient,
 {
-    pub fn new(joint_names: Vec<String>, shared_client: C) -> Self {
+    /// # Generate Partial Joint Client
+    ///
+    /// This function check partial joint name and full joint name.
+    /// Only allow unique partial joint name and joint name contained full.
+    ///
+    /// # Important point
+    ///
+    /// Partial Joint name is changed to dictionary order.
+    ///
+    pub fn new(joint_names: Vec<String>, shared_client: C) -> Result<Self, Error> {
+        use std::collections::HashSet;
+
+        // check length between full and partial
         let full_joint_names = shared_client.joint_names().to_vec();
-        Self {
+        if joint_names.len() > full_joint_names.len() {
+            return Err(Error::LengthMismatch {
+                model: full_joint_names.len(),
+                input: joint_names.len(),
+            });
+        }
+
+        // check redundant joint name of partial
+        let input_len = joint_names.len();
+        let mut unique_joint_names = joint_names
+            .clone()
+            .into_iter()
+            .collect::<HashSet<String>>()
+            .into_iter()
+            .collect::<Vec<String>>();
+        unique_joint_names.sort();
```

Comment:
Is it necessary to sort?
And, is it necessary to be Vec? (HashSet also has `len` method.)

Reply:
I forget to eliminate sort. It is Relics of the past.
Changing Vec is same.
In preview version, unique is necessary for Partial